torch/fx/experimental/Partitioner.py

```diff
@@ -0,0 +1,151 @@
+        self.partitions: List[Partition] = []
+
+    """
+    After a partitioner is created,
+    it first check if multiple types of devices (backends) are involved
+    So far, we assume that there is only one backend and one device with unlimited memory
+    """
+
+    def partition_graph(self, fx_module: GraphModule, devices: List[str]) -> None:
+        graph: Graph = fx_module.graph
+        """check if devices are different backend types"""
+        multiple_backend_types: bool = self.check_multiple_backend_types(devices)
+        """one node could be partitioned into multiple partitions
+           mapping {node_index: [partition_ids]}
+        """
+        node_to_partition_mapping: Dict[int, List[int]] = {}
+        """only one type of devices"""
+        if not multiple_backend_types:
+            node_to_partition_mapping = self.load_balanced_partition(graph, devices)
+        else:
+            """todo: multiple_type_of_devices"""
+            pass
+        self.do_partition(fx_module, node_to_partition_mapping)
+        return
+
+    def load_balanced_partition(self, graph: Graph, devices: List[str]) -> Dict[int, List[int]]:
+        """same type of devices"""
+        node_to_partition_mapping: Dict[int, List[int]] = {}
+        """only one device, so far, assume memory is unlimited. Otherwise, memory size should be checked"""
+        if len(devices) == 1:
+            node_to_partition_mapping = self.find_single_partition(graph, devices[0])
+        else:
+            """todo: multiple devices of same type"""
+            pass
+        return node_to_partition_mapping
+
+    def find_single_partition(self, graph: Graph, device: str) -> Dict[Node, List[int]]:
+        """only one partition (one graph on one device)"""
+        partition_0: Partition = self.create_partition(0)
+        node_to_partition_mapping: Dict[int, List[int]] = {}
+        """mapping all graph nodes into partition_0"""
+        for i, node in enumerate(graph.nodes):
+            node_to_partition_mapping[i] = [0]
+            partition_0.add_node(i, node.name, node.op)
+
+        return node_to_partition_mapping
+
+    def do_partition(self, fx_module: GraphModule, node_to_partition_mapping: Dict[int, List[int]]) -> None:
+        """mark the partition on each node in the fx_module"""
+        graph = fx_module.graph
+        for i in range(len(graph.nodes)):
+            graph.nodes[i].partition_ids = node_to_partition_mapping[i]
+        return
+
+    def dump_partition_DAG(self) -> None:
+        print('dumping partition ...\n')
+        for partition in self.partitions:
+            print(partition)
+        return
+
+    def create_partition(self, parition_id: int) -> Partition:
+        """create a partition and append it to self.partitions"""
+        partition: Partition = Partition(parition_id)
+        self.partitions.append(partition)
+        return partition
+
+    def check_multiple_backend_types(self, devices: List[str]) -> bool:
```

Comment:
We needed this in the glow partitioner but we can get rid of it for this partitioner.